### PR TITLE
Made it not crash and neatened code

### PR DESCRIPTION
**Pull request by Injector-Spenral**
Made it not crash, and neatened the code.
NOTE: The maths doesn't work, it doesn't return a correct distance.

### CODE REVIEW

**Review by LucienMorey — DISMISSED** [review]

File: vision.py [hunk 1]
```diff
@@ -224,17 +215,17 @@ def run(self):
...
+                distance, angle = results
                 self.Connection.send_results(
-                    (results[2], results[1], time.monotonic())
+                    (distance, angle, time.monotonic())
                 )  # distance (meters), angle (radians), timestamp
 
                 if self.zooming == True:
                     self.lastZoom = self.zoom
-                    self.zoom = self.translate(abs(results[1]), 0.45, 0, 100, 200)
+                    self.zoom = self.translate(abs(angle), 0.45, 0, 100, 200)
```
Comment (LucienMorey) [inline vision.py]: What are the magic numbers?

File: vision.py [hunk 2]
```diff
@@ -154,43 +132,51 @@ def find_power_port(self, frame: np.ndarray) -> tuple:
...
+
+    def get_middles(self, contour: np.ndarray) -> tuple:
+        """ Use the cv2 moments to find the centre x and y of the contour. """
         M = cv2.moments(contour)
+        if M["m00"] != 0:
         cX = int(M["m10"] / M["m00"])
-        cY = int(M["m01"] / M["m00"])
+        else:
+            cX = 160
+        # cY = int(M["m01"] / M["m00"])
```
Comment (LucienMorey) [inline vision.py]: Does this need to be here anymore?

File: vision.py [hunk 3]
```diff
@@ -224,17 +215,17 @@ def run(self):
                 self.CameraManager.sinks[0].getError()
             )
         else:
-
+            self.frame = cv2.flip(self.frame, -1)
```
Comment (LucienMorey) [inline vision.py]: Is -1 the magic number for flipping the way you want it flipped?

File: vision.py (hunk 2 above)
Comment (raulvera59) [reply vision.py]: Consider replacing this with a comment to the effect that we are after the center of the hexagon, not of the shape we see.

File: vision.py (hunk 1 above)
Comment (Injector-Spenral) [reply vision.py]: I killed zoom.

File: vision.py (hunk 2 above)
Comment (Injector-Spenral) [reply vision.py]: We use the bottom of the hexagon.

File: vision.py (hunk 3 above)
Comment (Injector-Spenral) [reply vision.py]: Yes.

**Review by LucienMorey — APPROVED** [review]

**Pull request merged**